src/pymoca/backends/casadi/generator.py

```diff
@@ -355,7 +366,18 @@ def exitForEquation(self, tree):
             indexed_symbols_full = []
             for k in indexed_symbols:
                 s = f.indexed_symbols[k]
-                orig_symbol = self.nodes[self.current_class][s.tree.name]
+                if k.name() in [x.name for x in self.model.delayed_states]:
+                    # We are missing a similarly shaped delayed symbol. Replace/make it.
```

Comment:
We should probably change things such that the symbol in `self.model.delayed_states` has the right dimension for the for loop.